rasa/nlu/classifiers/logistic_regression_classifier.py

```diff
@@ -0,0 +1,182 @@
+import logging
+from typing import Any, Text, Dict, List, Type
+
+from joblib import dump, load
+from scipy.sparse import hstack, vstack, csr_matrix
+from sklearn.linear_model import LogisticRegression
+
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.engine.recipes.default_recipe import DefaultV1Recipe
+from rasa.engine.graph import ExecutionContext, GraphComponent
+from rasa.nlu.featurizers.featurizer import Featurizer
+from rasa.nlu.classifiers.classifier import IntentClassifier
+from rasa.nlu.classifiers import LABEL_RANKING_LENGTH
+from rasa.shared.nlu.training_data.training_data import TrainingData
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import TEXT, INTENT
+
+
+logger = logging.getLogger(__name__)
+
+
+@DefaultV1Recipe.register(
+    DefaultV1Recipe.ComponentType.INTENT_CLASSIFIER, is_trainable=True
+)
+class LogisticRegressionClassifier(IntentClassifier, GraphComponent):
+    """Intent classifier using the Logistic Regression."""
+
+    @classmethod
+    def required_components(cls) -> List[Type]:
+        """Components that should be included in the pipeline before this component."""
+        return [Featurizer]
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
+        return ["sklearn"]
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """The component's default config (see parent class for full docstring)."""
+        return {"class_weight": "balanced", "max_iter": 100, "solver": "lbfgs"}
+
+    def __init__(
+        self,
+        config: Dict[Text, Any],
+        name: Text,
+        model_storage: ModelStorage,
+        resource: Resource,
+    ) -> None:
+        """Construct a new classifier."""
+        self.name = name
+        self.clf = LogisticRegression(
```

Comment:
Would we want to expose more parameters than these? E.g. `random_state` would be useful for reproducibility. `solver` seems a very technical detail, and afaik there's some cross-dependency with the `penalty. Should in this case both be configurable, so users can also chose `l1` or `elasticnet`? Alternatively, could also pass on `**config` filtered for all valid parameter choices?

Reply:
Oh totally! The current PR is a bit more work in progress than "ready for review". I'll update the title to reflect this. 

At the moment I'm still running the benchmarks so that we can confirm that it's not an unreasonable alternative. 

In general, I think it's best to be explicit in the parameters that we allow. The **kwargs trick is nice for quick iteration, but requires deep knowledge of sklearn which can get in the way of using the component.

Reply:
I've just added the random seed but I'm wondering if it makes sense to either be a bit minimal in the parameters that we allow or if we want to allow the user to fully customize everything. 

@koernerfelicia @mleimeister got any strong opinions on this?

Reply:
Apart from allowing full customization (which I think we don't necessarily need), another argument for being explicit about all default parameters is transparency and ensuring reproducibility. If e.g. Sklearn would change a default, we wouldn't notice, but the component suddenly behaves differently. I think we list all defaults like this in `CountVectorsFeaturizer`, but some would rarely be set in the pipeline config.